src/ducks/Alert/components/AlertModalContent/StepsContent/NotificationAndPrivacy/FrequencySelector/FrequencySelector.js

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { useField, useFormikContext } from 'formik'
+import Select from '@santiment-network/ui/Search/Select/Select'
+import { Checkbox } from '@santiment-network/ui/Checkboxes/Checkboxes'
+import Input from '@santiment-network/ui/Input'
+import { getFrequencyPeriods, getFrequencyTypes } from './utils'
+import styles from './FrequencySelector.module.scss'
+
+function FrequencySelector () {
+  const { values } = useFormikContext()
+  const [, { value: cooldown }, { setValue: setCooldown }] = useField(
+    'cooldown'
+  )
+  const [, { value: isRepeating }, { setValue: setIsRepeating }] = useField(
+    'isRepeating'
+  )
+
+  const cooldownPeriod = cooldown.slice(-1)
+  const cooldownCount = cooldown.slice(0, cooldown.length - 1)
+
+  const frequencyTypes = useMemo(
+    () => getFrequencyTypes(values.settings.metric),
+    []
+  )
+  const [type, setType] = useState(frequencyTypes[0])
+  const frequencyPeriods = useMemo(() => getFrequencyPeriods(type), [type])
+  const cooldownFrequencyPeriod = frequencyPeriods.find(
+    ({ value }) => value === cooldownPeriod
+  )
```

Comment:
Better to memoize it using `useMemo` otherwise it will do the `O(n)` lookup on every rerender